### ISSUE TITLE
Support LaTeX inside HTML / components

### ISSUE THREAD

**Issue by nvlang** [pngwn/MDsveX#560]
### Problem

```mdx
<span>$f$</span> $f$
```

Yields "`$f$` $f$". Similarly, math inside custom components also won't get rendered (this is actually my use case — I was trying to render math inside a `<Callout>` component).

I suspect this could be fixed by messing with the order in which different parts of the markdown file are processed, but I'm not familiar with MDsveX's internals, so this is merely an uneducated guess.

### Setup

```js
// File: mdsvex.config.js
import { defineMDSveXConfig } from 'mdsvex';
import rehypeMathjax from 'rehype-mathjax';
// import rehypeKatex from 'rehype-katex-svelte';
import remarkMath from 'remark-math'; // remark-math@3.0.1

export default defineMDSveXConfig({
    remarkPlugins: [remarkMath],
    rehypePlugins: [rehypeKatex]
    // rehypePlugins: [rehypeMathjax]
});
```

### Workaround

A workaround (though IMO rather suboptimal):

```svelte
<!-- File: M.svelte -->
<script lang="ts">
import katex from 'katex';
export let math: string = '';
</script>

{@html katex.renderToString(math)}
```

```mdx
<span><M math="f"/></span> $f$
```

### Quick Tangent

I get the following error at `rehypePlugins: [rehypeKatex]`:

```
Type 'Plugin<[(KatexOptions | undefined)?], Root, Root>' is not assignable to type 'Plugin | [Plugin, Settings]'.
  Type 'Plugin<[(KatexOptions | undefined)?], Root, Root>' is not assignable to type 'Plugin'.
    Type 'void | Transformer<Root, Root> | undefined' is not assignable to type 'void | Transformer<Node, Node> | undefined'.
      Type 'Transformer<Root, Root>' is not assignable to type 'void | Transformer<Node, Node> | undefined'.
        Type 'Transformer<Root, Root>' is not assignable to type 'Transformer<Node, Node>'.
          Type 'Node' is not assignable to type 'Root'.
```

Or, if using `rehypePlugins: [rehypeMathjax]`:

```
Type '(options?: Readonly<Options> | null | undefined) => (tree: Root) => undefined' is not assignable to type 'Plugin | [Plugin, Settings]'.
  Type '(options?: Readonly<Options> | null | undefined) => (tree: Root) => undefined' is not assignable to type 'Plugin'.
    Type '(tree: Root) => undefined' is not assignable to type 'void | Transformer<Node, Node> | undefined'.
      Type '(tree: Root) => undefined' is not assignable to type 'Transformer<Node, Node>'.
        Types of parameters 'tree' and 'tree' are incompatible.
          Property 'children' is missing in type 'Node' but required in type 'Root'.
```

### Addendum

This is probably more of a feature request than a bug report. I hope that this repo is at least the right place to submit it to (i.e., that it's not a bug in `remark-math` or whatnot). In any event, thank you for a great tool!

Also, I care a great deal about supporting math expressions, and would be more than happy to try to help with some contributions. However, I read that substantial changes are planned for v1, so I don't know what the best place to start would be.

**Comment by GuillaumeQuenneville** [comment]
Math issues are tracked here https://github.com/pngwn/MDsveX/issues/302
I think the last comments have a solution to your problem.

**Comment by nvlang** [comment]
> Math issues are tracked here #302
> I think the last comments have a solution to your problem.

Thank you for the hint! I felt like this was a different enough (in particular, more specific) to warrant a separate issue, but I did read through #302 and the comments there, and did not find any solution to my issue (aside from the workaround that I already mentioned above).

**Comment by pngwn** [comment]
This coula slo be related to the fact that you can't have markdown 'in' HTML at the moment. STuff like this doesn't work:

```html
<span>*hi*</span>
```

Instead you need to have newline in between:

```html
<span>

*hi*

</span>
```

This is how markdown works in general but is a limitation I plan to address.

**Comment by nvlang** [comment]
@pngwn I don't know how I missed that! Would it be helpful to edit this issue to address "markdown in HTML/components" in general?

**Comment by pngwn** [comment]
Yeah, that would be great! I thought we had an issue for this but I could not find it, so renaming this would be perfect, thank you.

**Comment by Leftium** [comment]
I think my feature request is related to this issue. I wish there was syntax that allowed transforming bits of markdown inside (Svelte/Svx) templates:

Ideally like Svelte's `{@html}` [syntax](https://svelte.dev/docs/svelte/@html): 
```
<span>Please see the {@svx [about](/about)} page.<span>
```

Or even as a [template tag function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates):
```
<script>
  import {svx} from 'mdsvex';
  const html = svx`[about](/about)`
</script>
```
